addons/mass_mailing/models/mass_mailing.py

```diff
@@ -444,6 +445,29 @@ def _get_clicks_ratio(self, cr, uid, ids, name, arg, context=None):
             res[record['id']] = 100 * record['nb_clicks'] / record['nb_mails']
 
         return res
+
+    @api.multi
+    def _unsubscribe_token(self, res_id, email):
+        """Generate a secure hash for this mailing list and parameters.
+
+        This is appended to the unsubscription URL and then checked at
+        unsubscription time to ensure no malicious unsubscriptions are
+        performed.
+
+        :param int res_id:
+            ID of the resource that will be unsubscribed.
+
+        :param str email:
+            Email of the resource that will be unsubscribed.
+        """
+        icp = self.env["ir.config_parameter"]
+        salt = icp.get_param("mass_mailing.salt")
+        if not salt:
+            salt = uuid4()
+            icp.set_param("mass_mailing.salt", salt)
```

Comment:
You should not be creating a new ICP entry for this. There's a special ICP key named `database.secret` that can be used for such purposes. It's properly protected from read access from regular users (which something you had overlooked your PR), so you should make sure to use sudo() mode to read it, so that `_unsubscribe_token()` can be called by an unprivileged user.